File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/CreateBigSegmentStoreIntegrationExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class CreateBigSegmentStoreIntegrationExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var integrationDeliveryConfigurationPost = new IntegrationDeliveryConfigurationPost();
        integrationDeliveryConfigurationPost.config(JSON.deserialize("""
            {
                "optional": "example value for optional formVariables property for sample-integration",
                "required": "example value for required formVariables property for sample-integration"
            }
        """, Map.class));
        integrationDeliveryConfigurationPost.on(false);
        integrationDeliveryConfigurationPost.name("Example persistent store integration");
        integrationDeliveryConfigurationPost.tags(List.of (
            "example-tag"
        ));

        try
        {
            var response = new PersistentStoreIntegrationsBetaApi(config).createBigSegmentStoreIntegration(
                "projectKey_string", // projectKey
                "environmentKey_string", // environmentKey
                "integrationKey_string", // integrationKey
                integrationDeliveryConfigurationPost
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling PersistentStoreIntegrationsBetaApi#createBigSegmentStoreIntegration");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/CreateBigSegmentStoreIntegrationExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class CreateBigSegmentStoreIntegrationExample
{
    fun createBigSegmentStoreIntegration()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val integrationDeliveryConfigurationPost = IntegrationDeliveryConfigurationPost(
            config = Serializer.moshi.adapter<Map<String, Any>>().fromJson("""
                {
                    "optional": "example value for optional formVariables property for sample-integration",
                    "required": "example value for required formVariables property for sample-integration"
                }
            """)!!,
            on = false,
            name = "Example persistent store integration",
            tags = listOf (
                "example-tag",
            ),
        )

        try
        {
            val response = PersistentStoreIntegrationsBetaApi().createBigSegmentStoreIntegration(
                projectKey = null,
                environmentKey = null,
                integrationKey = null,
                integrationDeliveryConfigurationPost = integrationDeliveryConfigurationPost,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling PersistentStoreIntegrationsBetaApi#createBigSegmentStoreIntegration")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling PersistentStoreIntegrationsBetaApi#createBigSegmentStoreIntegration")
            e.printStackTrace()
        }
    }
}
